apis/pipelines/v1alpha3/object_hasher.go

```diff
@@ -0,0 +1,47 @@
+package v1alpha3
+
+import (
+	"crypto/sha1"
+	"hash"
+	"sort"
+)
+
+var hashFieldSeparator = []byte{0}
+
+//+kubebuilder:object:generate=false
```

Comment:
This file has been moved into the apis subdirectory to avoid the following cyclic dependency:

` apis (pipeline spec) -> controller (object hasher) -> apis (NamedValue)`

Given that ObjectHasher is only used from within the apis package, there is no need for it to be in controllers.

Reply:
the annotation is needed because without it `controller-gen` would generate a `DeepCopy` function for it